runtime/cudaq/platform/default/rest/helpers/oqc/OQCServerHelp.cpp

```diff
@@ -0,0 +1,298 @@
+  return std::string(env_var);
+}
+
+// Check if a key exists in the backend configuration
+bool OQCServerHelper::keyExists(const std::string &key) const {
+  return backendConfig.find(key) != backendConfig.end();
+}
+
+std::vector<std::string> OQCServerHelper::getJobID(int n) {
+  RestHeaders headers = OQCServerHelper::getHeaders();
+  nlohmann::json j;
+  std::vector<std::string> output;
+  for (int i = 0; i < n; ++i) {
+    nlohmann::json_v3_11_1::json response = client.post(
+        backendConfig.at("url"), backendConfig.at("job_path"), j, headers);
+    output.push_back(response[0]);
+  }
+  return output;
+}
+
+std::string OQCServerHelper::makeConfig(int shots) {
+  return "{\"$type\": \"<class 'scc.compiler.config.CompilerConfig'>\", "
+         "\"$data\": {\"repeats\": " +
+         std::to_string(shots) +
+         ", \"repetition_period\": null, \"results_format\": {\"$type\": "
+         "\"<class 'scc.compiler.config.QuantumResultsFormat'>\", \"$data\": "
+         "{\"format\": {\"$type\": \"<enum "
+         "'scc.compiler.config.InlineResultsProcessing'>\", \"$value\": 1}, "
+         "\"transforms\": {\"$type\": \"<enum "
+         "'scc.compiler.config.ResultsFormatting'>\", \"$value\": 3}}}, "
+         "\"metrics\": {\"$type\": \"<enum "
+         "'scc.compiler.config.MetricsType'>\", \"$value\": 6}, "
+         "\"active_calibrations\": [], \"optimizations\": {\"$type\": \"<class "
+         "'scc.compiler.config.Tket'>\", \"$data\": {\"tket_optimizations\": "
+         "{\"$type\": \"<enum 'scc.compiler.config.TketOptimizations'>\", "
+         "\"$value\": 30}}}}}";
+}
+
+// Create a job for the OQC quantum computer
+ServerJobPayload
+OQCServerHelper::createJob(std::vector<KernelExecution> &circuitCodes) {
+  // Check if the necessary keys exist in the configuration
+  if (!keyExists("target") || !keyExists("qubits") || !keyExists("job_path"))
+    throw std::runtime_error("Key doesn't exist in backendConfig.");
+  std::vector<ServerMessage> jobs(circuitCodes.size());
+  std::vector<std::string> task_ids =
+      OQCServerHelper::getJobID(static_cast<int>(circuitCodes.size()));
+
+  for (size_t i = 0; i < circuitCodes.size(); ++i) {
+    nlohmann::json j;
+    j["tasks"] = std::vector<nlohmann::json>();
+    // Construct the job message
+    nlohmann::json job;
+    job["task_id"] = task_ids[i];
+    job["config"] = makeConfig(static_cast<int>(shots));
+    job["program"] = circuitCodes[i].code;
+    j["tasks"].push_back(job);
+    jobs[i] = j;
+  }
+
+  // Return a tuple containing the job path, headers, and the job message
+  return std::make_tuple(backendConfig.at("url") +
+                             backendConfig.at("job_path") + "/submit",
+                         getHeaders(), jobs);
+}
+
+// From a server message, extract the job ID
+std::string OQCServerHelper::extractJobId(ServerMessage &postResponse) {
+  // If the response does not contain the key 'id', throw an exception
```

Comment:
This comment may not be valid anymore. Seems like we've added a code path to handle this condition at the `Executor` level?

Reply:
It looks like this comment was removed in the latest commit, so I will mark this conversation as closed.